packages/math/tools/unicode-xml-to-sile.xsl

```diff
@@ -0,0 +1,201 @@
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+<!--
+  Stylesheet to convert the unicode.xml file to a SILE Lua file:
+  xsltproc unicode-xml-to-sile.xsl unicode.xml > ../packages/math/unicode-symbols-generated.lua
+  Where unicode.xml is:
+  https://raw.githubusercontent.com/w3c/xml-entities/gh-pages/unicode.xml
```

Comment:
Is the data we need not available anywhere in UCD/CLDR? It seems like this should be in lua-cldr not here. I'm working on fixing this up with build targets so it can be regenerated for now, but this doesn't feel like the right home.

Reply:
Bah. I did some digging an this xml-entities mapping seems to be maintained on it's own, and referenced in the MathML spec but not kept as part of UCD. Why do spec oversight orgs make things so convoluted? Bah.

Reply:
The W3C xml-entities repository mixes several sources (HTML entities, UCD and Unicode Math Technical Record, MathML specs, TeX / AMS / unicode-math names etc. ) into a single XML document.
It is used (according to Carlisle, the main editor, if I read that correctly) to prepare some of the tables in the MathML specs and elsewhere.
It's some sort of aggregation of specs from those various groups, converted (with lots of scripts and/or XSLT stylesheets) into one common file -- So it's a "derived" source indeed, but I found it neat to get most information directly, rather than doing the "assembly" work from all those scattered committee-designed sources ourselves, or rather than just "copying" some incomplete part of LaTeX's unicode-math's mappings (what SILE's code was doing, without providing a path for automating it).

Reply:
>  I'm working on fixing this up with build targets so it can be regenerated for now

Maybe wait -- I'll rebase this PR onto master (probably some small conflicts with the ssty / primes stuff on tests) and work on accent support, but I haven't checked yet that our Lua conversion was including all the necessary details (I did use some property to extract the combining classes, but didn't check it yet). We might need to map things slightly differently (as top/bottom and wide/regular accents). 
Eventually when we have it right, I don't expect these files to be regenerated often -- it's supposed to be fairly stable, though I did found some issue (now fixed by Carlisle).

Reply:
By the way, it obviously has some overlaps with our `char-def.lua` file used for linebreaking (notably), and if I understand it correctly, "stolen" from ConTeXt by Simons years ago. I haven't checked how those could be reconciled...

Reply:
Yes, and I agree what you have is much better than not having any automation at all. While I still might move *where* we put the generated data, generating it from relevant canonical sources (aggregated ones or otherwise) is clearly the right move.

Reply:
https://github.com/sile-typesetter/sile/issues/2176